pkg/oc/lib/newapp/app/imagestreamlookup.go

```diff
@@ -124,23 +126,44 @@ func (r ImageStreamSearcher) Search(precise bool, terms ...string) (ComponentMat
 				// using a "stable" branch by giving the control over version to the
 				// image stream author.
 				finalTag := searchTag
-				if specTag, ok := stream.Spec.Tags[searchTag]; ok && followTag && !specTag.HasAnnotationTag(imageapi.TagReferenceAnnotationTagHidden) {
+				var specTag *imagev1.TagReference
+				for _, t := range stream.Spec.Tags {
+					if specTag.Name != searchTag {
+						continue
+					}
+
+					specTag = &t
+					break
+				}
+
+				if specTag != nil && followTag && !HasAnnotationTag(specTag, imageapi.TagReferenceAnnotationTagHidden) {
 					if specTag.From != nil && specTag.From.Kind == "ImageStreamTag" && !strings.Contains(specTag.From.Name, ":") {
-						if destSpecTag, ok := stream.Spec.Tags[specTag.From.Name]; ok && !destSpecTag.HasAnnotationTag(imageapi.TagReferenceAnnotationTagHidden) {
-							if imageapi.LatestTaggedImage(stream, specTag.From.Name) != nil {
+						for _, destTag := range stream.Spec.Tags {
+							if destTag.Name != specTag.From.Name {
+								continue
+							}
+
+							if HasAnnotationTag(&destTag, imageapi.TagReferenceAnnotationTagHidden) {
+								break
+							}
+
+							// TODO(juanvallejo): we're no longer using an api helper here - but are duplicating logic
+							// from internal image api helper to make it suitable for the v1 api. Is this okay?
```

Comment:
Image controller will eventually need the external versions. Make the helpers in image package in that case. At some point in time when we'll decide to split oc out we'll have to figure something out.